src/setup_payload/tests/TestQRCode.cpp

```diff
@@ -117,54 +117,73 @@ void TestPayloadBase38Rep(nlTestSuite * inSuite, void * inContext)
 
 void TestBase38(nlTestSuite * inSuite, void * inContext)
 {
-    uint8_t input[] = { 10, 10, 10 };
+    uint8_t input[]                 = { 10, 10, 10 };
+    constexpr size_t kEncodedBufLen = 32;
+    char encodedBuf[kEncodedBufLen];
 
     // basic stuff
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 0).empty());
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 1) == "A0");
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 2) == "OT10");
-    NL_TEST_ASSERT(inSuite, base38Encode(input, 3) == "-N.B0");
+    base38Encode(input, 0, encodedBuf, kEncodedBufLen);
```

Comment:
The test is written expecting still the std::string output, which is not correct. Null-termination is missing from impl, so the strcmp below will fail. Did you run the tests and succeed?